opennms-doc/guide-admin/src/asciidoc/text/minion/jms.adoc

```diff
@@ -0,0 +1,81 @@
+
+// Allow GitHub image rendering
+:imagesdir: ../../images
+
+=== Using JMS
+
+By default, _{opennms-product-name}_ uses the _JMS_ protocol with an _ActiveMQ_ broker to communicate with _Minions_.
+This is used for both issuing remote procedure calls (RPCs, ie. ping this host) and for transporting unsolicited messages such as SNMP traps and syslog messages.
+_{opennms-product-name}_ provides an embedded _ActiveMQ_ broker to help simplify installation.
+
+
+TIP: It is also possible (and recommended for large installations) to use an external an external broker.
+
+==== Tuning the ActiveMQ broker
+
+The settings for the embedded _ActiveMQ_ broker are found in `$OPENNMS_HOME/etc/opennms-activemq.xml`.
+Memory and storage limits are conservative by default and should be tuned to accommodate your workload.
+Consider increasing the `memoryUsage` (defaults to 20MB) to `512MB` or greater, assuming you have enough heap available.
+
+TIP: If the memory limit is reached, https://activemq.apache.org/producer-flow-control.html[flow control] will prevent messages
+from being published to the broker.
+
+==== Monitoring the ActiveMQ broker using the Karaf shell
+
+The `opennms-activemq:stats` command available via the _Karaf shell_ can be used to show statistics about the embedded broker:
+
+[source]
+----
+opennms-activemq:stats
+----
+
+TIP: If the command is not available, try installing the feature using `feature:install opennms-activemq-shell`
+
+This command reports some high level broker statistics as well as message, enqueue and dequeue counts for the available queues.
+Pay close attention to the memory usage that is reported.
+If the usage is high, use the queue statistics to help isolate which queue is consuming most of the memory.
+
+The `opennms-activemq:purge-queue` command can be used to delete all of the available messages in a particular queue:
+
+[source]
+----
+opennms-activemq:purge-queue OpenNMS.Sink.Trap
+----
+
+==== Tuning the RPC client in OpenNMS
+
+The following system properties can be used to tune the thread pool used to issue RPCs:
+
+====== General
+[options="header, autowidth"]
+|===
+| Name                              | Default  | Description
```

Comment:
Can't see these properties getting used in code. Do we need these limits ?

Reply:
These are hidden in the context here: https://github.com/OpenNMS/opennms/blob/develop/core/ipc/rpc/jms-impl/src/main/resources/META-INF/opennms/applicationContext-rpc-client-jms.xml#L57

Reply:
Got it. I thought it's for the thread pool you created in this PR.